### DISCUSSION TITLE
API Post error 500 Language matching query does not exist

### DISCUSSION THREAD

**Discussion by demonoys** [WeblateOrg/weblate#6049] [Q&A] [answered]
Hi,

I'm trying to import some components in Weblate 4.5.3 (self hosted). I have a zip archive with multiple xliff files (base language is english) which i'm trying to import using the POST /api/projects/(string:project)/components/ api call, but I receive an internal server error. After digging through weblate's logs I found an error saying that "Language matching query does not exist".

Importing the zip file directly from the web UI works perfectly.
Anyone know what could be the cause of this issue?

**Comment by nijel** [comment, marked as answer]
There has been several improvements for this API endpoint in 4.6, try upgrading your Weblate first.
